### DISCUSSION TITLE
register reference update

### DISCUSSION THREAD

**Discussion by ikibalnyi** [react-hook-form/react-hook-form#2389] [Q&A]
Hi,

I wonder, why `defaultValues` is specified as a dependency for `register`?
What is the need to update the reference based on `defaultValues`?

https://github.com/react-hook-form/react-hook-form/blob/master/src/useForm.ts#L1146

**Comment by bluebill1049** [comment]
it's for `reset`.

**Reply by ikibalnyi** [reply]
hm, but how are they connected? I don't get

**Reply by bluebill1049** [reply]
because during `reset` invoking, we remove all input reference and we need to register with the latest `defaultValues`.
